python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -264,42 +352,92 @@ def _create_streaming_chat_message_content(
             metadata=metadata,
         )
 
-    def _format_chat_history(self, chat_history: ChatHistory) -> list[ChatRequestMessage]:
-        """Format the chat history to the expected objects for the client.
+    # endregion
+
+    def _get_metadata_from_response(self, response: ChatCompletions | AsyncStreamingChatCompletions) -> dict[str, Any]:
+        """Get metadata from the response.
 
         Args:
-            chat_history: The chat history.
+            response: The response from the service.
 
         Returns:
-            A list of formatted chat history.
+            A dictionary containing metadata.
         """
-        chat_request_messages: list[ChatRequestMessage] = []
+        return {
+            "id": response.id,
+            "model": response.model,
+            "created": response.created,
+            "usage": response.usage,
+        }
 
-        for message in chat_history.messages:
-            if message.role != AuthorRole.USER or not any(isinstance(item, ImageContent) for item in message.items):
-                chat_request_messages.append(_MESSAGE_CONVERTER[message.role](content=message.content))
-                continue
+    def _verify_function_choice_behavior(self, settings: AzureAIInferenceChatPromptExecutionSettings, **kwargs: Any):
+        """Verify the function choice behavior."""
+        kernel = kwargs.get("kernel", None)
+        arguments = kwargs.get("arguments", None)
+
+        if settings.function_choice_behavior is not None:
+            if kernel is None:
+                raise ServiceInvalidExecutionSettingsError("Kernel is required for tool calls.")
+            if arguments is None and settings.function_choice_behavior.auto_invoke_kernel_functions:
+                raise ServiceInvalidExecutionSettingsError("Kernel arguments are required for auto tool calls.")
+            if settings.extra_parameters is not None and settings.extra_parameters.get("n", 1) > 1:
```

Comment:
Does it make sense to throw here, or log a warning and set the `n` to 1 to continue the flow as is done in the OpenAI case?

Reply:
The OpenAI connector throws too in this case.